src/sdk/pycli/nnicli/nni_client.py

```diff
@@ -92,48 +72,384 @@ def _create_process(cmd):
             print(output.decode('utf-8').strip())
     return process.returncode
 
-def start_nni(config_file):
-    """start nni experiment with specified configuration file"""
-    cmd = 'nnictl create --config {}'.format(config_file).split(' ')
-    if _create_process(cmd) != 0:
-        raise RuntimeError('Failed to start nni.')
-
-def stop_nni():
-    """stop nni experiment"""
-    cmd = 'nnictl stop'.split(' ')
-    if _create_process(cmd) != 0:
-        raise RuntimeError('Failed to stop nni.')
-
-def version():
-    """return version of nni"""
-    return _nni_rest_get(VERSION_PATH, 'text')
-
-def get_experiment_status():
-    """return experiment status as a dict"""
-    return _nni_rest_get(STATUS_PATH)
-
-def get_experiment_profile():
-    """return experiment profile as a dict"""
-    return _nni_rest_get(EXPERIMENT_PATH)
-
-def get_trial_job(trial_job_id):
-    """return trial job information as a dict"""
-    assert trial_job_id is not None
-    return _nni_rest_get(os.path.join(TRIAL_JOBS_PATH, trial_job_id))
-
-def list_trial_jobs():
-    """return information for all trial jobs as a list"""
-    return _nni_rest_get(TRIAL_JOBS_PATH)
-
-def get_job_statistics():
-    """return trial job statistics information as a dict"""
-    return _nni_rest_get(JOB_STATISTICS_PATH)
-
-def get_job_metrics(trial_job_id=None):
-    """return trial job metrics"""
-    api_path = METRICS_PATH if trial_job_id is None else os.path.join(METRICS_PATH, trial_job_id)
-    return _nni_rest_get(api_path)
-
-def export_data():
-    """return exported information for all trial jobs"""
-    return _nni_rest_get(EXPORT_DATA_PATH)
+def _check_endpoint(endpoint):
+    if endpoint is None:
+        raise RuntimeError("This instance hasn't been connect to an experiment.")
+
+class NNITrialResult:
+    """
+    NNITrialResult stores the result information of a trial job.
+
+    Parameters
+    ----------
+    json_obj:
+        json object that stores the result information
+
+    Attributes
+    ----------
+    parameter:
+        hyper parameters for this trial
+    value:
+        final result
+    id:
```

Comment:
what is the relation between this id and `trialJobId` in `NNITrialMetricData`?

Reply:
I follow the keys defined in the api, here they should be same value.